From e002696b0673dff7ec9584e00b1546d2f850121c Mon Sep 17 00:00:00 2001
From: Christoph Cullmann <cullmann@kde.org>
Date: Mon, 16 May 2022 19:47:04 +0200
Subject: [PATCH] Revert "fix crash on file browser activation"

This reverts commit a28624c877cfe6db6ae6ab257f95af36213b25e0.

=> this fix is not enough, the kconfigGroup is just not valid anymore here
in some cases

BUG: 453795
---
 addons/filebrowser/katefilebrowserplugin.cpp | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/addons/filebrowser/katefilebrowserplugin.cpp b/addons/filebrowser/katefilebrowserplugin.cpp
index 6df25239a..f03c78698 100644
--- a/addons/filebrowser/katefilebrowserplugin.cpp
+++ b/addons/filebrowser/katefilebrowserplugin.cpp
@@ -100,12 +100,9 @@ void kateFileBrowserPluginView::createFileBrowser(bool visible)
 
     disconnect(m_toolView.get(), SIGNAL(toolVisibleChanged(bool)), this, SLOT(createFileBrowser(bool)));
     m_fileBrowser = std::make_unique<kateFileBrowser>(m_mainWindow, m_toolView.get());
-
-    // read config only if there yet, see bug 453234
-    if (cg) {
-        m_fileBrowser->readSessionConfig(*cg);
-        cg.reset();
-    }
+    m_fileBrowser->readSessionConfig(*cg);
+    // delete the config now
+    cg.reset();
 }
 
 bool kateFileBrowserPluginView::eventFilter(QObject *obj, QEvent *event)
-- 
GitLab

